### ISSUE TITLE
Check for unused parameters

### ISSUE THREAD

**Issue by NormanTUD** [tensorflow/tfjs#5798]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow.js version (you are using): Latest
- Are you willing to contribute it (Yes/No): yes

**Describe the feature and the current behavior/state.**

Since on Keras, parameters are like `kernel_initializer' and on tfjs parameters are in Camel-case, it is sometimes confusing if the parameter has been set correctly. I suggest console.log warnings for unused layer-(and possibly everything else)-options, so that when you enter "kernel_initializer" instead of "kernelInitializer", it will tell you that the parameter "kernel_initializer" was ignored. This would make debugging programs in tfjs way more easy.

**Will this change the current api? How?**

No.

**Who will benefit with this feature?**

Anybody who has experience with Keras in Python and now tries to do things the tfjs-way.

**Any Other info.**

I've modified the tf.js for the dense-layer in this way, just as a test if this would work:

```

	  function Dense(args) {
	    var _this4;
	    var args_keys = Object.keys(args);

	    _this4 = _Layer2.call(this, args) || this; // Default activation: Linear (none).

	    _this4.activation = null;
	    _this4.useBias = true;
	    _this4.kernel = null;
	    _this4.bias = null;
	    _this4.DEFAULT_KERNEL_INITIALIZER = 'glorotNormal';
	    _this4.DEFAULT_BIAS_INITIALIZER = 'zeros';

	    if (args.batchInputShape == null && args.inputShape == null && args.inputDim != null) {
	      // This logic is copied from Layer's constructor, since we can't
	      // do exactly what the Python constructor does for Dense().
	      var batchSize = null;

	      if (args.batchSize != null) {
	        batchSize = args.batchSize;
	      }

	      _this4.batchInputShape = [batchSize, args.inputDim];
	    }

	    _this4.units = args.units;
	    assertPositiveInteger(_this4.units, 'units');
	    _this4.activation = getActivation(args.activation);

	    if (args.useBias != null) {
	      _this4.useBias = args.useBias;
	    }

	    _this4.kernelInitializer = getInitializer(args.kernelInitializer || _this4.DEFAULT_KERNEL_INITIALIZER);
	    _this4.biasInitializer = getInitializer(args.biasInitializer || _this4.DEFAULT_BIAS_INITIALIZER);
	    _this4.kernelConstraint = getConstraint(args.kernelConstraint);
	    _this4.biasConstraint = getConstraint(args.biasConstraint);
	    _this4.kernelRegularizer = getRegularizer(args.kernelRegularizer);
	    _this4.biasRegularizer = getRegularizer(args.biasRegularizer);
	    _this4.activityRegularizer = getRegularizer(args.activityRegularizer);
	    _this4.supportsMasking = true;
	    _this4.inputSpec = [{
	      minNDim: 2
	    }];
	    var layer_keys = Object.keys(_this4);
	    for (var layer_keys_counter = 0; layer_keys_counter != layer_keys.length; layer_keys_counter++) {
		    var this_key = layer_keys[layer_keys_counter];
		    var this_key_index = args_keys.indexOf(this_key);
		    if (this_key_index != -1) {
			args_keys.splice(this_key_index, 1);
		    }
	    }
	    if(args_keys.length) {
		    console.log("Unused args for dense layer:", args_keys);
	    }
	    return _this4;
	  }
```

For `var dense = tf.layers.dense({units: 10, activation: 'softmax', bla: 20});` this now reports:

`Unused args for dense layer: Array [ "bla" ]`, which was exactly what I wanted, since "bla" is not a valid key.

This method relies on the idea that the layer-object's names are the same as the parameter names. I have not checked whether this is true for all functions or not, so there may be more work that results from this as it seems, but it would be *really* useful.

**Comment by rthadur** [comment]
cc @pyu10055 @lina128

**Comment by pyu10055** [comment]
@NormanTUD As you stated, providing warning message to list out the wrong keys will be very useful to avoid confusing between the keras and tfjs naming. It will be even better to be able to suggest the right naming for the user. 
Seem you are willing to contribute this feature, we are happy to review your PRs. Thanks

**Comment by NormanTUD** [comment]
I need some time for this, but I see 2 ways of realizing this. The first is to have an array of valid values, which would be hard to maintain. The other one, which is less accurate, but really easy to implement and maintain I think, is to check the args for the existence of a "_" character. A warning like "did you mean kernelInitializer instead of kernel_initializer?" would also be easily automated (just replace /_(.)/ with uppercase \1). Do you think this would be OK?

**Comment by gaikwadrahul8** [comment]
Hi, @NormanTUD 

Apologize for the delayed response and we are re-visiting our older feature requests and checking whether those feature requests implemented or not as of now and you shown interest to submit PR to take care of this feature request so May I know are you working on it or Have you submitted PR for this feature in TFJs please? 

If someone wants to contribute for this feature then you're always welcome and please feel free to do and please refer these links [Ref-1](https://github.com/tensorflow/tfjs/blob/master/CONTRIBUTING.md), [Ref-2](https://github.com/tensorflow/tfjs/blob/master/CONTRIBUTING_MISSING_OP.md) . Thank you!
